sbncode/SinglePhotonAnalysis/HelperFunctions/helper_PandoraPFParticles.cxx

```diff
@@ -0,0 +1,192 @@
+#include "sbncode/SinglePhotonAnalysis/HelperFunctions/helper_PandoraPFParticles.h"
+
+namespace single_photon
+{
+  PandoraPFParticle::PandoraPFParticle( 
+      art::Ptr<recob::PFParticle> input_PFParticle,
+      std::vector< art::Ptr< larpandoraobj::PFParticleMetadata > > input_MetaData,
+      std::vector< art::Ptr<recob::Vertex > > input_Vertex,
+      std::vector< art::Ptr<recob::Cluster> > input_Clusters,
+      std::vector< art::Ptr<recob::Shower > > input_Showers,
+      std::vector< art::Ptr<recob::Track  > > input_Tracks,
+      art::FindManyP<recob::Hit>   input_Hits
+      )
+    :
+      pPFParticle(input_PFParticle),
+      pMetaData(input_MetaData),
+      pVertex(input_Vertex),
+      pClusters(input_Clusters)
+  {
+    pPFParticleID = pPFParticle->Self();
+    pPdgCode = pPFParticle->PdgCode();
+
+    //Get recob::Shower/Track
+    const unsigned int nShowers(input_Showers.size());
+    const unsigned int nTracks(input_Tracks.size());
+    if(nShowers+nTracks != 1) mf::LogDebug("SinglePhoton") << "  No just one shower/track is associated to PFParticle " << pPFParticleID << "\n";
+
+    pHasShower = nShowers;
+    pHasTrack = nTracks;
+    if(pHasShower == 1) pShower=input_Showers.front();
+    if(pHasTrack == 1) pTrack=input_Tracks.front();
```

Comment:
No need to compare a boolean to anything.
```suggestion
    if(pHasShower) pShower=input_Showers.front();
    if(pHasTrack) pTrack=input_Tracks.front();
```

Reply:
`pHasShower` and `pHasTrack` are designed to be integers for the case that there are more then just one tracks/showers for a given `PFParticle`, which has not been seen yet; currently, it would only work if a `PFParticle` is associated to only one shower or only one track.